pkg/lib/record_transaction_detail_test.go

```diff
@@ -112,3 +113,116 @@ func TestTransactionDetailOutputWithContinuationRecords(t *testing.T) {
+ */
+func TestTransactionDetailOutputWithContinuationRecords_AdHocContinuationFormat(t *testing.T) {
+	data := `16,266,1912,,GI2118700002010,20210706MMQFMPU8000001,Outgoing Wire Return,-/
+88,CREF: 20210706MMQFMPU8000001/
+88,EREF: 20210706MMQFMPU8000001/
+88,DBIC: GSCRUS33/
+88,CRNM: ABC Company/
+88,DBNM: SAMPLE INC./`
+
+	record := transactionDetail{}
+
+	size, err := record.parse(data)
+	require.NoError(t, err)
+
+	require.Equal(t, "266", record.TypeCode)
+	require.Equal(t, "1912", record.Amount)
+	require.Equal(t, "", string(record.FundsType.TypeCode))
+	require.Equal(t, "", record.FundsType.Date)
+	require.Equal(t, "", record.FundsType.Time)
+	require.Equal(t, "GI2118700002010", record.BankReferenceNumber)
+	require.Equal(t, "20210706MMQFMPU8000001", record.CustomerReferenceNumber)
+	require.Equal(t, "Outgoing Wire Return", record.Text)
+	require.Equal(t, 73, size)
+
+	result := record.string()
+	// NB. This is the current output of a Detail formatted like the above.
+	expectResult := `16,266,1912,,GI2118700002010,20210706MMQFMPU8000001,Outgoing Wire Return/`
+	require.Equal(t, expectResult, result)
+	require.Equal(t, len(expectResult), len(result))
+}
+
+/**
+ * This test outlines the behavior of a Detail record when the Detail includes Identifier and Text records
+ * that include a Terminal character ('/'). The BAI2 spec indicates that most fields "must not contain a comma or a slash"
+ * and places strict rules on when/where a slash can occur. Nevertheless, real world data is observed where a slash
+ * is used in records that disallow it.
+ *
+ * This library currently raises an error when parsing a record that includes slash characters.
+ */
+func TestTransactionDetailOutput_FailsIfRecordIncludesIllegalCharacters(t *testing.T) {
+	data := `16,447,928650,,SPB2322684598521,AB/GS/RPFILERP0001/RPBA0001,ACH Credit Payment,Entry Description: TRADE; -, SEC: CTX, Client Ref ID: AB/GS/TEST0001/RPBA0001, GS ID: SPB2322684598521/
+88,EREF: AB/GS/RPFILERP0001/RPBA0001/
+88,DBNM: SAMPLE INC/
+88,CACT: ACHCONTROLOUTUSD01/`
+
+	record := transactionDetail{}
+
+	size, err := record.parse(data)
+	require.Equal(t, fmt.Errorf("TransactionDetail: unable to parse Text"), err)
+	require.Equal(t, size, 0)
+
+	// require.Equal(t, "447", record.TypeCode)
+	// require.Equal(t, "928650", record.Amount)
+	// require.Equal(t, "", string(record.FundsType.TypeCode))
+	// require.Equal(t, "", record.FundsType.Date)
+	// require.Equal(t, "", record.FundsType.Time)
+	// require.Equal(t, "SPB2322684598521", record.BankReferenceNumber)
+	// require.Equal(t, "AB/GS/RPFILERP0001/RPBA0001", record.CustomerReferenceNumber)
+	// require.Equal(t, "ACH Credit Payment", record.Text)
+	// require.Equal(t, 79, size)
+
+	// result := record.string()
+	// // NB. This is the current output of a Detail formatted like the above.
+	// expectResult := `16,447,928650,,SPB2322684598521,AB/GS/RPFILERP0001/RPBA0001,ACH Credit Payment/`
+	// require.Equal(t, expectResult, result)
```

Comment:
I'm fine allowing `/` if it's showing up in real world data. Let the real world work.